dotCMS/src/main/java/com/dotmarketing/portal/struts/DotPortletAction.java

```diff
@@ -907,40 +928,17 @@ protected static void _checkUserPermissions(Inode webAsset, User user, int permi
 		if (!InodeUtils.isSet(webAsset.getInode()))
 			return;
 		if (!perAPI.doesUserHavePermission(webAsset, permission, user)) {
-			Logger.debug(DotPortletAction.class, "_checkUserPermissions: user does not have permissions ( " + permission + " ) over this asset: " + webAsset);
-			List<Role> rolesPublish = perAPI.getRoles(webAsset.getInode(), PermissionAPI.PERMISSION_PUBLISH, "CMS Owner", 0, -1);
-			List<Role> rolesWrite = perAPI.getRoles(webAsset.getInode(), PermissionAPI.PERMISSION_WRITE, "CMS Owner", 0, -1);
-			
-			Role cmsOwner;
-			try {
-				cmsOwner = APILocator.getRoleAPI().loadCMSOwnerRole();
-			} catch (DotDataException e) {
-				Logger.error(DotPortletAction.class,e.getMessage(),e);
-				throw new ActionException(e);
-			}
-			boolean isCMSOwner = false;
-			if(rolesPublish.size() > 0 || rolesWrite.size() > 0){
-				for (Role role : rolesPublish) {
-					if(role.getId().equals(cmsOwner.getId())){
-						isCMSOwner = true;
-						break;
-					}
-				}
-				if(!isCMSOwner){
-					for (Role role : rolesWrite) {
-						if(role.getId().equals(cmsOwner.getId())){
-							isCMSOwner = true;
-							break;
-						}
-					}
-				}
-				if(!isCMSOwner){
-					throw new ActionException(WebKeys.USER_PERMISSIONS_EXCEPTION);
-				}
-			}else{
-				throw new ActionException(WebKeys.USER_PERMISSIONS_EXCEPTION);
-			}	
-					
+			_checkUserPermissions(webAsset.getInode(), permission);
+		}
+	}
+
+	protected static void _checkUserPermissions(final Folder folder, final User user, final int permission) throws ActionException, DotDataException {
```

Comment:
not sure if we should use underscore convention, you can rename to internalCheckUserPermissions or so

Reply:
I overloaded an existing method. For that reason, Id rather prefer to keep the same convention that has been used so far